src/osp/Resource/AssetImporter.cpp

```diff
@@ -67,20 +67,23 @@ void osp::AssetImporter::load_sturdy_file(std::string const& filepath, Package&
         std::cout << "Error: couldn't open file: " << filepath << "\n";
     }
 
-    load_sturdy(gltfImporter, pkg);
+    // repurpose filepath into unique identifier for file resources
+    std::string dataResourceName = filepath.substr(0, filepath.find('.')) + ':';
+
+    load_sturdy(gltfImporter, dataResourceName, pkg);
 
     gltfImporter.close();
 }
 
 void osp::AssetImporter::load_part(TinyGltfImporter& gltfImporter,
-    Package& pkg, unsigned id)
+    Package& pkg, unsigned id, std::string_view resPrefix)
```

Comment:
Please change the type of id from "unsigned" to, well, anything else. e.g. std::uint64_t or something.

Reply:
In this case specifically, or in general? Cap used `unsigned` for a lot of things like this, I assume because it's clearer. If you use `uint64_t` or `uint32_t` it sort of falsely implies that there's a specific reason to use that size, when in reality it's just supposed to be a small positive integer and the size doesn't matter.

Reply:
the size of unsigned changes based on the platform being compiled for. No one wants that.

Specific bit sizes result in the least amount of surprise.

Reply:
> If you use uint64_t or uint32_t it sort of falsely implies that there's a specific reason to use that size, when in reality it's just supposed to be a small positive integer and the size doesn't matter.

That's solved by using a typedef, and a comment explaining that the choice was arbitrary.

Reply:
In this case the IDs are all very small (like, almost guaranteed to be <100 and usually <10) so changing sizes doesn't matter but a typedef will work too. I propose we make a new PR for this though, as this is just one of many instances of its use in the repo

Reply:
Sure that's fine.

If they're always smaller than 256, then use std::uint8_t.